### PR TITLE
TOML declarative config for `mod_sic` and `mod_time`

### PR DESCRIPTION
**Pull request by leszke**

### CODE REVIEW

File: test/config_parser_SUITE.erl [hunk 1]
```diff
@@ -2587,7 +2587,10 @@ mod_shared_roster_ldap(_Config) ->
     ?errf(T(#{<<"ldap_filter">> => 1})).
 
 mod_sic(_Config) ->
-    check_iqdisc(mod_sic).
+    check_iqdisc(mod_sic),
+    T = fun(Opts) -> #{<<"modules">> => #{<<"mod_sic">> => Opts}} end,
+    M = fun(Cfg) -> modopts(mod_sic, Cfg) end,
+    ?eqf(M([]), T(#{})).
```
Comment (chrzaszcz) [inline test/config_parser_SUITE.erl:2593]: This check is not doing much, but it's not harmful, we can keep it.

File: test/config_parser_SUITE.erl [hunk 2]
```diff
@@ -2616,7 +2616,10 @@ mod_stream_management_stale_h(_Config) ->
     ?errf(T(#{<<"geriatric">> => <<"one">>})).
 
 mod_time(_Config) ->
-    check_iqdisc(mod_time).
+    check_iqdisc(mod_time),
+    T = fun(Opts) -> #{<<"modules">> => #{<<"mod_time">> => Opts}} end,
+    M = fun(Cfg) -> modopts(mod_time, Cfg) end,
+    ?eqf(M([]), T(#{})).
```
Comment (chrzaszcz) [inline test/config_parser_SUITE.erl:2622]: This check is not doing much, but we can keep it, it's no harm.

**Review by chrzaszcz — COMMENTED** [review]
Minor comments. This PR requires rebasing on top of `toml-config` - the rest is fine!

File: test/config_parser_SUITE.erl (hunk 1 above)
Comment (leszke) [reply test/config_parser_SUITE.erl:2593]: Correct, I thought it won't hurt to test empty config, particularly if specification allows it.

**Review by chrzaszcz — APPROVED** [review]
Looks good!

**Pull request merged**